### ISSUE TITLE
Primary key and foreign key with different number length

### ISSUE THREAD

**Issue by derickgn** [rsim/oracle-enhanced#1831]
### Steps to reproduce
Simply create an run two migrations, first one with only and id, second one with a id and a reference to the first. (I've used rails generate model script to do this)

Example:

```ruby
class CreateTableA < ActiveRecord::Migration[5.2]
  def change
    create_table :table_a do |t|
      t.timestamps
    end
  end
end
```

```ruby
class CreateTableB < ActiveRecord::Migration[5.2]
  def change
    create_table :table_b do |t|
      t.references :table_a
      t.timestamps
    end
  end
end
```

### Expected behavior
Both id and table_a_id should have they data_type set to number(38,0)

### Actual behavior
In Oracle database you'll see that the id columns will have data_type set to number(38,0) and table_a_id column set to number(19,0)

### System configuration
**Rails version**:5.2.2

**Oracle enhanced adapter version**:5.2.4

**Ruby version**:6.0.1

**Oracle Database version**:12.1

**Comment by yahonda** [comment]
Thanks for opening an issue. It may be related to https://github.com/rails/rails/commit/14db455156c , I mean we may need to implement this logic into Oracle enhanced adapter whose integer is "larger" than bigint. Will investigate.

**Comment by yahonda** [comment]
@derickgn Would you update your Gemfile and see if the issue has been addressed or not?

```ruby
gem "activerecord-oracle_enhanced-adapter", github: "yahonda/oracle-enhanced", branch: "diag1831_release52"
```

**Comment by yahonda** [comment]
Oracle enhanced adapter 5.2.5 has been released including this fix #1832.

**Comment by derickgn** [comment]
> @derickgn Would you update your Gemfile and see if the issue has been addressed or not?
> 
> ```ruby
> gem "activerecord-oracle_enhanced-adapter", github: "yahonda/oracle-enhanced", branch: "diag1831_release52"
> ```

Yes, it has been resolved!
Thank you very much.
